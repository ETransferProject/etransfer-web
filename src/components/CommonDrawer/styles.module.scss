@import '/src/assets/theme/color.scss';
@import '/src/assets/theme/font.scss';
@import '/src/styles/variable.scss';

.common-drawer {
  min-width: $page-min-width;
  :global {
    .ant-drawer-content-wrapper {
      border-radius: 8px 8px 0px 0px;
      background-color: $page-bg;
    }
    .ant-drawer-content {
      border-radius: 8px 8px 0px 0px;
      background-color: $page-bg;
    }
    .ant-drawer-header {
      flex: none;
      height: 56px;
      padding: 15px 16px;
      border: none;
      .ant-drawer-header-title {
        text-align: center;
        position: relative;
        .ant-drawer-title {
          color: $primary-text;
          font-size: 16px;
          line-height: 24px;
          font-weight: $font-family-medium-weight;
          font-family: $font-family-medium-family;
        }
        .ant-drawer-close {
          position: absolute;
          right: 0;
          margin: 0;
        }
      }
    }
    .ant-drawer-body {
      overflow-y: auto;
      padding: 0 16px 16px;
    }
  }
}
