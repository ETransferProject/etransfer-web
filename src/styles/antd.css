html {
  --ant-primary-color: #285eff;
  --ant-primary-color-hover: #3b6cff;
  --ant-primary-color-active: #1e52f0;
  --ant-success-color: #00b73e;
  --ant-error-color: #f53f3f;
  --ant-error-color-hover: #fc7373;
  --ant-warning-color-active: #cf3737;
  --ant-warning-color: #f90;
  --ant-info-color: #285eff;
}
