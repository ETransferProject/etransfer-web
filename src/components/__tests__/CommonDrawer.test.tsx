import { render, screen } from '@testing-library/react';
import { vi, describe, expect, test, beforeEach } from 'vitest';
import CommonDrawer from '../CommonDrawer';
import { Drawer } from 'antd';
import clsx from 'clsx';

// Mock Antd Drawer
vi.mock('antd', async (importOriginal) => {
  const actual = await importOriginal<typeof import('antd')>();
  return {
    ...actual,
    Drawer: vi.fn((props) => <div data-testid="mock-drawer" {...props} />),
  };
});

// Mock clsx
vi.mock('clsx', () => ({
  default: vi.fn(() => 'mocked-class'),
}));

describe('CommonDrawer Component', () => {
  const defaultProps = {
    open: true,
    onClose: vi.fn(),
  };

  beforeEach(() => {
    vi.clearAllMocks();
  });

  test('renders with default props', () => {
    render(<CommonDrawer {...defaultProps} />);

    // Check if the Drawer component is rendered
    const drawer = screen.getByTestId('mock-drawer');
    expect(drawer).toBeInTheDocument();
    expect(Drawer).toHaveBeenCalledWith(
      expect.objectContaining({
        placement: 'bottom',
        width: '100%',
        height: '88%',
        zIndex: 300,
        destroyOnClose: true,
        closable: true,
      }),
      expect.anything(),
    );
  });

  test('merges class names correctly', () => {
    render(<CommonDrawer {...defaultProps} className="custom-class" />);

    // Check if the class names are merged correctly
    expect(clsx).toHaveBeenCalledWith(expect.stringMatching(/common-drawer/), 'custom-class');
  });

  test('passes custom props to antd Drawer', () => {
    const customProps = {
      width: '50%',
      height: '50%',
      zIndex: 500,
      placement: 'right' as const,
    };

    render(<CommonDrawer {...defaultProps} {...customProps} />);

    // Check if the custom props are passed correctly
    expect(Drawer).toHaveBeenCalledWith(expect.objectContaining(customProps), expect.anything());
  });

  test('uses default values when props not provided', () => {
    render(<CommonDrawer {...defaultProps} width={undefined} height={undefined} />);

    // Check if the default values are used
    expect(Drawer).toHaveBeenCalledWith(
      expect.objectContaining({
        width: '100%',
        height: '88%',
      }),
      expect.anything(),
    );
  });

  test('matches snapshot', () => {
    const { asFragment } = render(<CommonDrawer {...defaultProps} />);

    // Check if the snapshot matches the rendered component
    expect(asFragment()).toMatchSnapshot();
  });
});
