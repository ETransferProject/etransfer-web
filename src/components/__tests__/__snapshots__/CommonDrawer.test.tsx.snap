// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CommonDrawer Component > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="mocked-class"
    closeicon="[object Object]"
    data-testid="mock-drawer"
    height="88%"
    open=""
    placement="bottom"
    width="100%"
    zindex="300"
  />
</DocumentFragment>
`;
